### ISSUE TITLE
Sonic Prequel Crackups: Text formatting error.

### ISSUE THREAD

**Issue by waspennator** [ruffle-rs/ruffle#6087]
### Describe the bug

Text isn't properly formatted to fit inside textboxes, resulting in text going out of the boxes or other text issues. Skip past the intro, start watching and the issues should pop up as early as the first short.
https://www.newgrounds.com/portal/view/572498
![Sonic Crackup error](https://user-images.githubusercontent.com/78768499/150060923-5e54b624-e569-4cb4-a7ef-2cc99460170c.png)

### Expected behavior

The text should be fitting inside the box completely without trailing out of it, like this pic of it running on flash.
![Sonic Crackup proper](https://user-images.githubusercontent.com/78768499/150061463-47829bb4-d79d-4895-97db-ac1afc1b0142.png)

### Affected platform

Browser's extension

### Operating system

Windows 11

### Browser

Edge 97

**Comment by waspennator** [comment]
Still an issue as of the latest nightly.

**Comment by waspennator** [comment]
Still happening as of the latest nightly.

**Comment by kjarosh** [comment]
This SWF uses device fonts, see https://github.com/ruffle-rs/ruffle/wiki/Frequently-Asked-Questions-For-Users#how-do-i-change-the-fonts-used-by-flash-on-my-website

**Comment by waspennator** [comment]
Also found another problem, for some reason the S in the sega logo doesnt look similar to the ega part

**Comment by kjarosh** [comment]
> Also found another problem, for some reason the S in the sega logo doesnt look similar to the ega part

That's a bug in the SWF, not Ruffle. In FP it looks the same

**Comment by waspennator** [comment]
Yeah, jumped the gun cause one of the video references on YouTube of the swf, the creator fixed the s for it

**Comment by waspennator** [comment]
Finally fixed by #22254 when enabled, guess ill leave it open until it gets enabled by default

**Comment by waspennator** [comment]
@kjarosh 
Formatting is close now testing on Firefox with canvas fonts, but still a little off.

<img width="1359" height="958" alt="Image" src="https://github.com/user-attachments/assets/9c5baa78-e529-4f07-88f6-ae198efee03f" />

**Comment by kjarosh** [comment]
For the record, this is what it looks like for me on Flash Player, Windows:

<img alt="Image" src="https://github.com/user-attachments/assets/ec1bcf9c-4bf5-4b05-81f9-e2eeed74f860" />

The remaining issue is heavily device-specific, as it depends on the exact font provided by the OS for `Gill Sans`. IIUC the original font is available only on macos. In the example above, Windows returns a serif font that is relatively narrow and makes the text fit the bubble.

You can try installing the original Gill Sans font and using the canvas font renderer or FreeType.

Finishing canvas font renderer _might_ fix it, but I wouldn't bet on it, as it really depends on the OS, not Ruffle.

For content preservation, the best course of action would be to provide the web version of this font or a compatible one to Ruffle, as you cannot really guarantee that someone will have this font installed. This is possible without the canvas font renderer, but the renderer doesn't allow providing own fonts yet.

**Comment by waspennator** [comment]
> For the record, this is what it looks like for me on Flash Player, Windows:
> 
> ![Image](https://private-user-images.githubusercontent.com/1507072/655408270-ec1bcf9c-4bf5-4b05-81f9-e2eeed74f860.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3OTAwMjc2ODQsIm5iZiI6MTc5MDAyNzM4NCwicGF0aCI6Ii8xNTA3MDcyLzY1NTQwODI3MC1lYzFiY2Y5Yy00YmY1LTRiMDUtODFmOS1lMmVlZWQ3NGY4NjAucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI2MDkyMSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNjA5MjFUMjE0OTQ0WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZGJkZDUyYmZkNDExMDE5NmE2MDUyOGJiNGIxYjYzODAyNmM3YmNkODFhODNjZWIwZWRjODE4MGY3OTQ0Yzc0NCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmcmVzcG9uc2UtY29udGVudC10eXBlPWltYWdlJTJGcG5nIn0.dSNwtGV91LOGjDRSV8brQEs39_2t9VFFLCz8wltI16o)
> The remaining issue is heavily device-specific, as it depends on the exact font provided by the OS for `Gill Sans`. IIUC the original font is available only on macos. In the example above, Windows returns a serif font that is relatively narrow and makes the text fit the bubble.
> 
> You can try installing the original Gill Sans font and using the canvas font renderer or FreeType.
> 
> Finishing canvas font renderer _might_ fix it, but I wouldn't bet on it, as it really depends on the OS, not Ruffle.
> 
> For content preservation, the best course of action would be to provide the web version of this font or a compatible one to Ruffle, as you cannot really guarantee that someone will have this font installed. This is possible without the canvas font renderer, but the renderer doesn't allow providing own fonts yet.

Trying it out on my windows laptop on ruffle web to check and it seems pretty much identical to flash using canvas, so im guessing the issue now is device os specific.
